decorator.rb

```diff
@@ -0,0 +1,31 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class BaseDecorator < Nameable
+  attr_reader :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize!
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    str = @nameable.correct_name
+    str.size <= 10 ? str : str[0...10]
```

Comment:
- [ ] When I run the code to trim the name (as shared in the project guideline) I get an error as a result of the `size`. This is because in the CapitalizeDecorator class the bang sign (`!`) makes the method return `nil`, hence by the time the program gets to TrimmerDecorator there is nothing to work with.

  Kindly remove the bang sign from `@nameable.correct_name.capitalize!` to fix this

  ![bertr-size](https://user-images.githubusercontent.com/36542986/162925210-63bca719-5a86-4ea5-a0d9-ef3d75ac8cdf.jpg)

Reply:
Thank you @TedLivist. This is fixed in the recent commit made.